packages/nx/src/nx-init/add-nx-to-nest.ts

```diff
@@ -0,0 +1,339 @@
+    webpackOptions,
+    entryFile,
+    plugins,
+    sourceRoot: nestCLIJson['sourceRoot'],
+  };
+}
+
+function addNestPluginToPackageJson(repoRoot: string) {
+  const path = joinPathFragments(repoRoot, `package.json`);
+  const json = readJsonFile(path);
+  json.devDependencies['@nrwl/nest'] = require('../../package.json').version;
+  writeJsonFile(path, json);
+}
+
+function createProjectJson(
+  repoRoot: string,
+  packageName: string,
+  nestCLIOptions: NestCLIOptions
+) {
+  const path = joinPathFragments(repoRoot, 'project.json');
+  const json = {
+    $schema: 'node_modules/nx/schemas/project-schema.json',
+    name: packageName,
+    sourceRoot: nestCLIOptions.sourceRoot,
+    projectType: 'application',
+    targets: {},
+    tags: [],
+  };
+
+  json.targets['serve'] = {
+    executor: '@nrwl/js:node',
+    options: {
+      buildTarget: `${packageName}:build`,
+    },
+  };
+
+  if (nestCLIOptions.webpackOptions) {
+    json.targets['build'] = {
+      executor: '@nrwl/webpack:webpack',
+      outputs: ['{options.outputPath}'],
+      options: {
+        target: 'node',
+        compiler: 'tsc',
+        outputPath: `dist/${packageName}`,
+        main: joinPathFragments(
+          nestCLIOptions.sourceRoot,
+          nestCLIOptions.entryFile
+        ),
+        tsConfig: 'tsconfig.build.json',
+      },
+      configurations: {
+        production: {
+          optimization: true,
+          extractLicenses: true,
+          inspect: false,
+        },
+      },
+    };
+    json.targets['serve'] = {
+      ...json.targets['serve'],
+      configurations: {
+        production: {
+          buildTarget: `${packageName}:build:production`,
+        },
+      },
+    };
+  } else {
+    json.targets['build'] = {
+      executor: '@nrwl/js:tsc',
+      outputs: ['{options.outputPath}'],
+      options: {
+        outputPath: `dist/${packageName}`,
+        main: joinPathFragments(
+          nestCLIOptions.sourceRoot,
+          nestCLIOptions.entryFile
+        ),
+        tsConfig: 'tsconfig.build.json',
+      },
+    };
+    json.targets['serve'] = {
+      ...json.targets['serve'],
+      configurations: {
+        debug: {
+          inspect: 'inspect',
+        },
+      },
+    };
+
+    // if we're using nrwl/js, then we add nrwl/js analyzeSourceFiles to nx.json
+    addNrwlJsPluginsConfig(repoRoot);
+  }
+
+  // lint
+  // TODO: nestjs cli uses parserOptions with __dirname -> .eslintrc.js is a requirement. What can we do?
+  json.targets['lint'] = {
+    executor: '@nrwl/linter:eslint',
+    outputs: ['{options.outputFile}'],
+    options: {
+      lintFilePatterns: ['src/**/*.ts', 'test/**/*.ts'],
+    },
+  };
+
+  writeJsonFile(path, json);
+}
+
+function addNrwlJsPluginsConfig(repoRoot: string) {
+  const path = joinPathFragments(repoRoot, 'nx.json');
+  const json = readJsonFile(path);
+
+  if (!json.pluginsConfig) {
+    json.pluginsConfig = {
+      '@nrwl/js': {
+        analyzeSourceFiles: true,
+      },
```

Comment:
Type this: `NrwlJsPluginConfig`

- This is in `packages/nx/src/project-graph/build-project-graph.ts`, it will need to be exported. I'd move it to `packages/nx/src/config/nx-json.ts` since the config is always in there.